contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -257,19 +343,173 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
         );
 
         // Burn all OETH
-        uint256 oethBalance = oeth.balanceOf(address(this));
-        IVault(vaultAddress).burnForStrategy(oethBalance);
+        uint256 oethToBurn = oeth.balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        // Get the strategy contract's ether balance.
+        // This includes all that was removed from the Curve pool and
+        // any ether that was sitting in the strategy contract before the removal.
+        uint256 ethBalance = address(this).balance;
+        // Convert all the strategy contract's ether to WETH and transfer to the vault.
+        weth.deposit{ value: ethBalance }();
+        require(
+            weth.transfer(vaultAddress, ethBalance),
+            "Transfer of WETH not successful"
+        );
+
+        emit Withdrawal(address(weth), address(lpToken), ethBalance);
+        emit Withdrawal(address(oeth), address(lpToken), oethToBurn);
+    }
+
+    /***************************************
+            Curve pool Rebalancing
+    ****************************************/
+
+    /**
+     * @notice Mint OTokens and one-sided add to the Curve pool.
+     * This is used when the Curve pool does not have enough OTokens and too many ETH.
+     * The OToken/Asset, eg OETH/ETH, price with increase.
+     * The amount of assets in the vault is unchanged.
+     * The total supply of OTokens is increased.
+     * The asset value of the strategy and vault is increased.
+     * @param _oTokens The amount of OTokens to be minted and added to the pool.
+     */
+    function mintAndAddOTokens(uint256 _oTokens)
+        external
+        onlyStrategist
```

Comment:
nit: I feel like Governor should be able to do everything that the Strategist can do

Reply:
I think I've flipped on this a number of times. We can pick this up in the new generalised AMO for frxETH